### PR TITLE
Ci update

### PR DESCRIPTION
**Pull request by 1Revenger1**
Update action versions so that github actions stop complaining about Node and set/get-ouput commands. ~~The only one that causes warnings now is `download-artifact` but that hasn't gotten updates for a while.~~ `download-artifact` was updated 2 hours ago so no more warnings.

### CODE REVIEW

**Comment by mdurrani808** [comment]
Didn't see this.
Closes #519

**Comment by mdurrani808** [comment]
Ah whoops. Misread node for npm, need more sleep. There is another issue, let me grab a link to a failed CI to show.

https://github.com/PhotonVision/photonvision/actions/runs/3293708288/jobs/5430467353
failed under photon-build (the first one).

**Comment by 1Revenger1** [comment]
I'm trying to replicate the error but I haven't been able to reproduce it. I removed the node container though (since we already have the setup-node action) so maybe that helped to fix it? Either that or I'm cursed to never run into the issue.

Going to test the jar to make sure it still works on my Pi4 then this should be good to go.

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -296,11 +296,12 @@ jobs:
...
       # Upload final fat jar as artifact.
-      - uses: actions/upload-artifact@master
+      - uses: actions/upload-artifact@v3
         with:
           name: jar
           path: photon-server/build/libs
-      - uses: actions/upload-artifact@master
+      - uses: actions/upload-artifact@v3
+        if: github.event_name != 'pull_request'
         with:
```
Comment (mcm001) [inline .github/workflows/main.yml:305]: Why not on PRs anymore?

Comment (1Revenger1) [reply .github/workflows/main.yml:305]: The image is not created on PRs, and it was causing a warning.

**Comment by gerth2** [comment]
I'm 99% sure this Closes #519

**Review by mcm001 — APPROVED** [review]

**Pull request merged**